File: Difficulty: Basic/Find number of days between two given dates/find-number-of-days-between-two-given-dates.py
#User function Template for python3

from datetime import date

class Solution:
    def noOfDays(self, d1, m1, y1, d2, m2, y2):
        # code here 
        date1=date(y1,m1,d1)
        
        date2=date(y2,m2,d2)
        
        return abs((date2-date1).days)


#{ 
 # Driver Code Starts
#Initial Template for Python 3

if __name__ == '__main__': 
    t = int (input ())
    for _ in range (t):
        d1,m1,y1=map(int,input().split())
        d2,m2,y2=map(int,input().split())
        
        ob = Solution()
        print(ob.noOfDays(d1,m1,y1,d2,m2,y2))
# } Driver Code Ends

File: Difficulty: Basic/Find number of days between two given dates/find-number-of-days-between-two-given-dates.cpp
//{ Driver Code Starts
#include <bits/stdc++.h>
using namespace std;

// } Driver Code Ends
class Solution {
  public:
    bool leapyear(int y){
        
        return ( (y%100 != 0 && y%4==0) || y%400 == 0);
    }
    int calDays(int d,int m,int y){
        
      int DM[]={0,31,28+leapyear(y),31,30,31,30,31,31,30,31,30,31};
      
      while(--y)d+=365+leapyear(y);
      
      return accumulate(DM,DM+m,d);
    }
    int noOfDays(int d1, int m1, int y1, int d2, int m2, int y2) {
        // code here
        return abs(calDays(d1,m1,y1) - calDays(d2,m2,y2));
        
    }
};

//{ Driver Code Starts.
int main() {
    int t;
    cin >> t;
    while (t--) {
        int d1,d2,m1,m2,y1,y2;
        
        cin>>d1>>m1>>y1;
        cin>>d2>>m2>>y2;

        Solution ob;
        cout << ob.noOfDays(d1,m1,y1,d2,m2,y2) << endl;
    
cout << "~" << "\n";
}
    return 0;
}
// } Driver Code Ends